src/platform/OpenThread/MdnsImpl.cpp

```diff
@@ -28,8 +29,13 @@ namespace Mdns {
 
 CHIP_ERROR ChipMdnsInit(MdnsAsyncReturnCallback initCallback, MdnsAsyncReturnCallback errorCallback, void * context)
 {
-    // Intentionally empty
-    return CHIP_NO_ERROR;
+    ReturnErrorOnFailure(ThreadStackMgr().SetSrpDnsCallbacks(initCallback, errorCallback, context));
+
+    uint8_t mac[8];
+    char hostname[kMdnsHostNameMaxSize + 1] = "";
+    MakeHostName(hostname, sizeof(hostname), app::MdnsServer::Instance().FillMAC(mac));
```

Comment:
I think the platform code should not access the app/server/ code. Could this be passed as an argument and simply ignored in other platforms?

Reply:
Yeah I know, nevertheless it seemed to be for me the simplest way that will not complicate much code. Passing mac, as an argument is not that easy and nice solution, as only one platform is going to use it (Thread). If we want to pass `mac` to `ChipMdnsInit` we would need to do add such argument to the `Discovery_ImplPlatform::Start()` and `Discovery_ImplPlatform::Init()` methods that are also used for other platforms. Moreover we would need to add `mac` to the `StartResolver`, as it also uses `Init()` and is called in many places where getting MAC address is a bit confusing (e.g. shell command handler). Apart from that `FillMAC` method is implemented in the app/server, so I believe lib/mdns modules like Resolver also should not be dependent from it. I believe the best solution would be to move FillMAC method somewhere outside of the app/server code and use it wherever it's needed.

Reply:
Moving `FillMAC` to `lib/mdns` is also a good solution :)